core/src/main/java/org/openstack4j/openstack/artifact/internal/BaseArtifactServiceImpl.java

```diff
@@ -36,27 +37,27 @@ protected BaseArtifactServiceImpl(ArtifactType artifactType) {
 
     protected <T> T list(Class<T> clazz) {
 
-        return get(clazz,uri("/artifacts/"+ artifactType.value())).execute();
+        return get(clazz,uri(ClientConstants.PATH_ARTIFACTS+ClientConstants.URI_SEP+ artifactType.value())).execute();
```

Comment:
To save a couple of characters maybe just 
```
import org.openstack4j.core.transport.ClientConstants.PATH_ARTIFACTS;
import org.openstack4j.core.transport.ClientConstants.PATH_ARTIFACTS;
``` 
and `get(clazz, PATH_ARTIFACTS, URI_SEP, artifact.value()).execute()`?

Would it also be possible to avoid the `clazz`and use the type we would expect, e.g. `get(ToscaTemplatesArtifacts.class, PATH_ARTIFACTS, URI_SEP, artifact.value()).execute();`
Same for other methods.